osbuild/buildroot.py

```diff
@@ -221,6 +221,7 @@ def run(self, argv, monitor, binds=None, readonly_binds=None):
             "--new-session",
             "--setenv", "PATH", "/usr/sbin:/usr/bin",
             "--setenv", "PYTHONPATH", "/run/osbuild/lib",
+            "--setenv", "PYTHONUNBUFFERED", "1",
```

Comment:
Yeah, Ideally we would avoid shelling out where possible and have well-defined wrappers for commonly used tools that also handle return values / exit codes for the respective tools. Maybe some tasks for a new-comer to get into osbuild.

Reply:
I was thinking that even when we do have to shell out, we would catch the output of the tool and pass it up in our logging framework as a blob. So it would not be interleaved with other things, and it would be up to the UI layer to interleave/present the things.

Reply:
Yeah, totally. Probably a simple wrapper around `subprocess.run` will do Anaconda has a bunch of those to integrate it into their logging.